### PR TITLE
Alerting: Implement SaveAndApplyDefaultConfig in the remote Alertmanager struct

### PR DESCRIPTION
**Pull request by santihernandezc**
This PR implements the `SaveAndApplyDefaultConfig()` method in the remote Alertmanager struct. This method sends the default Grafana Alertmanager configuration over to the remote Alertmanager.

### CODE REVIEW

File: pkg/services/ngalert/remote/alertmanager.go [hunk 1]
```diff
@@ -30,15 +31,17 @@ type stateStore interface {
...
-	metrics  *metrics.RemoteAlertmanager
-	orgID    int64
-	ready    bool
-	sender   *sender.ExternalAlertmanager
-	state    stateStore
-	tenantID string
-	url      string
+	decrypt           DecryptFn
+	defaultConfig     string
+	defaultConfigHash string
```
Comment (santihernandezc) [inline pkg/services/ngalert/remote/alertmanager.go:36]: Only these two fields are new.

File: pkg/services/ngalert/remote/alertmanager.go [hunk 2]
```diff
@@ -182,36 +198,45 @@ func (am *Alertmanager) checkReadiness(ctx context.Context) error {
...
 	}
 
-	// Decrypt the configuration before comparing.
+	// Send the configuration only if we need to.
+	if !am.shouldSendConfig(ctx, rawDecrypted) {
+		return nil
+	}
+
+	return am.sendConfiguration(ctx, string(rawDecrypted), config.ConfigurationHash, config.CreatedAt, config.Default)
+}
```
Comment (santihernandezc) [inline pkg/services/ngalert/remote/alertmanager.go:213]: This method works in the same way as before, I just extracted some lines to `am.decryptConfiguration()` and `am.sendConfiguration()`.

File: pkg/services/ngalert/remote/alertmanager_test.go [hunk 3]
```diff
@@ -353,8 +358,8 @@ func TestIntegrationRemoteAlertmanagerApplyConfigOnlyUploadsOnce(t *testing.T) {
 	{
 		require.NoError(t, store.Set(ctx, cfg.OrgID, "alertmanager", "silences", base64.StdEncoding.EncodeToString([]byte("abc123"))))
 		require.NoError(t, store.Set(ctx, cfg.OrgID, "alertmanager", "notifications", base64.StdEncoding.EncodeToString([]byte("abc123"))))
-		fakeConfig.ID = 30000000000000000
```
Comment (santihernandezc) [inline pkg/services/ngalert/remote/alertmanager_test.go:356]: We're not using the ID anymore when sending a configuration, so changing the ID field won't change the payload we send. This fix is unrelated to this PR, but not worth making a separate PR.

File: pkg/services/ngalert/remote/alertmanager_test.go [hunk 4]
```diff
@@ -31,19 +31,26 @@ import (
...
-const testNflog2 = "OgoqCgZncm91cDISEgoJcmVjZWl2ZXIyEgV0ZXN0MyoMCLSDkbAGEMvaofYCEgwIxJ+RsAYQy9qh9gI6CioKBmdyb3VwMRISCglyZWNlaXZlcjESBXRlc3QzKgwItIORsAYQy9qh9gISDAjEn5GwBhDL2qH2Ag=="
+	// Valid Alertmanager state base64 encoded.
+	testSilence1 = "lwEKhgEKATESFxIJYWxlcnRuYW1lGgp0ZXN0X2FsZXJ0EiMSDmdyYWZhbmFfZm9sZGVyGhF0ZXN0X2FsZXJ0X2ZvbGRlchoMCN2CkbAGEJbKrMsDIgwI7Z6RsAYQlsqsywMqCwiAkrjDmP7///8BQgxHcmFmYW5hIFRlc3RKDFRlc3QgU2lsZW5jZRIMCO2ekbAGEJbKrMsD"
+	testSilence2 = "lwEKhgEKATISFxIJYWxlcnRuYW1lGgp0ZXN0X2FsZXJ0EiMSDmdyYWZhbmFfZm9sZGVyGhF0ZXN0X2FsZXJ0X2ZvbGRlchoMCN2CkbAGEJbKrMsDIgwI7Z6RsAYQlsqsywMqCwiAkrjDmP7///8BQgxHcmFmYW5hIFRlc3RKDFRlc3QgU2lsZW5jZRIMCO2ekbAGEJbKrMsDlwEKhgEKATESFxIJYWxlcnRuYW1lGgp0ZXN0X2FsZXJ0EiMSDmdyYWZhbmFfZm9sZGVyGhF0ZXN0X2FsZXJ0X2ZvbGRlchoMCN2CkbAGEJbKrMsDIgwI7Z6RsAYQlsqsywMqCwiAkrjDmP7///8BQgxHcmFmYW5hIFRlc3RKDFRlc3QgU2lsZW5jZRIMCO2ekbAGEJbKrMsD"
+	testNflog1   = "OgoqCgZncm91cDESEgoJcmVjZWl2ZXIxEgV0ZXN0MyoMCIzm1bAGEPqx5uEBEgwInILWsAYQ+rHm4QE="
+	testNflog2   = "OgoqCgZncm91cDISEgoJcmVjZWl2ZXIyEgV0ZXN0MyoMCLSDkbAGEMvaofYCEgwIxJ+RsAYQy9qh9gI6CioKBmdyb3VwMRISCglyZWNlaXZlcjESBXRlc3QzKgwItIORsAYQy9qh9gISDAjEn5GwBhDL2qH2Ag=="
+)
```
Comment (santihernandezc) [inline pkg/services/ngalert/remote/alertmanager_test.go:49]: I just added these into a single `const(...)` block

**Review by stevesg — APPROVED** [review]

**Pull request merged**